assets/src/models/finishedDetour.ts

```diff
@@ -1,9 +1,29 @@
-import { array, Infer, type } from "superstruct"
+import { array, Infer, number, type } from "superstruct"
 import { FinishedDetour } from "./detour"
 import { StopData, stopsFromData } from "./stopData"
 
 export const FinishedDetourData = type({
   missed_stops: array(StopData),
+  route_segments: type({
+    before_detour: array(
+      type({
+        lat: number(),
+        lon: number(),
+      })
```

Comment:
**suggestion (non-blocking):** since this is repeated a few times you could pull it out into its own non-`export`ed value.